From 2e69d32eb38e20439e6726ae2293674d6487d6ea Mon Sep 17 00:00:00 2001
From: James Slagle <jslagle@redhat.com>
Date: Tue, 25 Mar 2014 08:34:29 -0400
Subject: [PATCH 3/5] Expose dnsmasq options

Adds a new parameter, NeutronDnsmasqOptions, to the overcloud template.

Allows the ability to set dnsmasq options for neutron dhcp agent. This
will allow us to configure mtu to be 1400 for tenant instances on the
overcloud.  This should help with poor network performance and vm's that
are just plain unreachable via ssh due to the GRE tunnel overhead.

The default here has been set to:
dhcp-option-force=26,1400

This is the recommended way to configure OpenStack with the Open vSwitch
plugin per:
http://docs.openstack.org/admin-guide-cloud/content/openvswitch_plugin.html

All the documentation I can find on the web (openstack-dev,
ask.openstack.org, etc), recommend applying this setting. Others have
reported slow vm performance as well, and this resolves that issue
(apparently anyway...we'd need to test).

Change-Id: If24326045987b5a484ba2f71f591092987966536
Partial-Bug: #1270646

Conflicts:
	overcloud-source.yaml
---
 overcloud-source.yaml | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/overcloud-source.yaml b/overcloud-source.yaml
index bba161e..37acb84 100644
--- a/overcloud-source.yaml
+++ b/overcloud-source.yaml
@@ -121,6 +121,10 @@ Parameters:
     Default: ''
     Description: If set, the public interface is a vlan with this device as the raw device.
     Type: String
+  NeutronDnsmasqOptions:
+    Default: 'dhcp-option-force=26,1400'
+    Description: Dnsmasq options for neutron-dhcp-agent. The default value here forces MTU to be set to 1400 to account for the gre tunnel overhead.
+    Type: String
   notcomputeImage:
     Type: String
     Default: overcloud-control
@@ -443,6 +447,8 @@ Resources:
         ovs_db: mysql://neutron:unset@localhost/ovs_neutron?charset=utf8
         service-password:
           Ref: NeutronPassword
+        dnsmasq-options:
+          Ref: NeutronDnsmasqOptions
       ceilometer:
         db: mysql://ceilometer:unset@localhost/ceilometer
         metering_secret: {Ref: CeilometerMeteringSecret}
-- 
1.9.0

